### PR TITLE
[docathon] example for reset!

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: base/libgit2/libgit2.jl [hunk 1]
```diff
@@ -490,6 +490,22 @@ set by `mode`:
...
+# Example
+```julia
+# fetch changes
+LibGit2.fetch(repo)
+isfile(joinpath(repo_path, our_file)) # will be false
+
+# fastforward merge the changes
+LibGit2.merge!(repo, fastforward=true)
+
+# because there was not any file we need to reset the branch
```
Comment (ararslan) [inline base/libgit2/libgit2.jl]: I'm not sure what you mean. Could you clarify this?

File: base/libgit2/libgit2.jl [hunk 2]
```diff
@@ -490,6 +490,22 @@ set by `mode`:
...
   3. `Consts.RESET_HARD` - move HEAD to `id`, reset the index to `id`, and discard all working changes.
 
+# Example
+```julia
+# fetch changes
+LibGit2.fetch(repo)
+isfile(joinpath(repo_path, our_file)) # will be false
+
+# fastforward merge the changes
+LibGit2.merge!(repo, fastforward=true)
```
Comment (tkelman) [inline base/libgit2/libgit2.jl]: you don't have to tell this what to merge?

Comment (kshyatt) [inline base/libgit2/libgit2.jl]: I copied it from `test/libgit2.jl` so apparently not...

Comment (tkelman) [inline base/libgit2/libgit2.jl]: what a strange method then...

Comment (kshyatt) [inline base/libgit2/libgit2.jl]: We have such sights to show you...

Comment (ararslan) [inline base/libgit2/libgit2.jl]: I wonder if a more obvious method is available, because this is kind of bizarre.

**Comment by kshyatt** [comment]
Is this ok if it passes CI?

**Review by ararslan — APPROVED** [review]

**Comment by kshyatt** [comment]
Are the CI fails my fault?

**Comment by tkelman** [comment]
Possibly but they looks like they're related to https://github.com/JuliaLang/julia/pull/22963 rather than this

File: base/libgit2/libgit2.jl [hunk 3]
```diff
@@ -490,6 +490,23 @@ set by `mode`:
   2. `Consts.RESET_MIXED` - default, move HEAD to `id` and reset the index to `id`.
   3. `Consts.RESET_HARD` - move HEAD to `id`, reset the index to `id`, and discard all working changes.
 
+# Example
```
Comment (tkelman) [inline base/libgit2/libgit2.jl]: we've pluralized all of these right?

**Comment by kshyatt** [comment]
Everybody happy?

**Comment by kshyatt** [comment]
<3 Thanks for all the reviews!

**Pull request merged**